packages/sign-client/test/events.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { getSdkError, parseUri } from "@walletconnect/utils";
+import "mocha";
+import SignClient from "../src";
+import {
+  expect,
+  initTwoClients,
+  testConnectMethod,
+  TEST_SIGN_CLIENT_OPTIONS,
+  deleteClients,
+  TEST_NAMESPACES,
+  TEST_REQUIRED_NAMESPACES,
+  TEST_EMIT_PARAMS
+} from "./shared";
+import { EngineTypes, PairingTypes, SessionTypes } from "@walletconnect/types";
+
+describe("Sign Client Events Validation", () => {
+  it("init", async () => {
+    const client = await SignClient.init(TEST_SIGN_CLIENT_OPTIONS);
+    expect(client).to.be.exist;
+  });
+
+  describe("Events", () => {
+    it("session_proposal", async () => {
+
+      const clients = await initTwoClients();
+      const { A, B } = clients;
+
+      const connectParams: EngineTypes.ConnectParams = {
+        requiredNamespaces: TEST_REQUIRED_NAMESPACES,
+        relays: undefined,
+        pairingTopic: undefined,
+      };
+
+      const approveParams: Omit<EngineTypes.ApproveParams, "id"> = {
+        namespaces: TEST_NAMESPACES,
+      };
+
+      const { uri, approval } = await A.connect(connectParams);
+
+      let pairingA: PairingTypes.Struct | undefined;
+      let pairingB: PairingTypes.Struct | undefined;
+
+      if (!uri) throw new Error("uri is missing");
+
+      const uriParams = parseUri(uri);
+
+      pairingA = A.pairing.get(uriParams.topic);
+      expect(pairingA.topic).to.eql(uriParams.topic);
+      expect(pairingA.relay).to.eql(uriParams.relay);
+
+      if (!pairingA) throw new Error("expect pairing A to be defined");
+
+      let sessionA: SessionTypes.Struct;
+      let sessionB: SessionTypes.Struct;
+
+      await Promise.all([
+        new Promise<void>((resolve, reject) => {
```

Comment:
Honestly, you probably don't need to specify return type `<void>` in tests

Reply:
It's return `<void>` otherwise `resolve()` expects a value